src/runtime/components/Calendar.vue

```diff
@@ -0,0 +1,174 @@
+<script lang="ts">
+import { tv, type VariantProps } from 'tailwind-variants'
+import type { CalendarRootProps, RangeCalendarRootProps, CalendarCellTriggerProps, CalendarRootEmits, RangeCalendarRootEmits } from 'radix-vue'
+import type { DateValue } from '@internationalized/date'
+import type { AppConfig } from '@nuxt/schema'
+import _appConfig from '#build/app.config'
+import theme from '#build/ui/calendar'
+
+const appConfig = _appConfig as AppConfig & { ui: { calendar: Partial<typeof theme> } }
+
+const calendar = tv({ extend: tv(theme), ...(appConfig.ui?.calendar || {}) })
+
+type CalendarVariants = VariantProps<typeof calendar>
+
+export type ModelValue<T extends boolean> = (T extends true ? RangeCalendarRootProps['modelValue'] : CalendarRootProps['modelValue']) | undefined
+
+export interface CalendarProps<T extends boolean> extends Omit<CalendarRootProps & RangeCalendarRootProps, 'modelValue' | 'defaultValue' | 'dir' | 'locale' | 'calendarLabel'> {
+  /**
+   * The element or component this component should render as.
+   * @defaultValue 'div'
+   */
+  as?: any
+  /**
+   * The color of the calendar
+   */
+  color?: CalendarVariants['color']
+  /**
+   * The size of the calendar
+   */
+  size?: CalendarVariants['size']
+  /**
+   * Is this a range calendar
+   */
+  range?: T
+  /**
+   * Show month controls
+   */
+  monthControls?: boolean
+  /**
+   * Show year controls
+   */
+  yearControls?: boolean
+  defaultValue?: ModelValue<T>
+  class?: any
+  ui?: Partial<typeof calendar.slots>
+}
+
+export interface CalendarEmits extends Omit<CalendarRootEmits & RangeCalendarRootEmits, 'update:modelValue'> {}
```

Comment:
You can remove the `Omit` here so the docs show the `update:modelValue` event.

Reply:
In the documentation, `update:modelValue` is already displayed, it is taken from `defineModel`.